### PR TITLE
Use hide_internals in __tuple__ generation + tests

### PR DESCRIPTION
**Pull request by thisisrandy**
Per #53

### CODE REVIEW

**Comment by biqqles** [comment]
Fantastic, thank you, it will take me a couple of days to have time to review thoroughly.

File: dataclassy/decorator.py [hunk 1]
```diff
@@ -21,7 +21,7 @@ def dataclass(cls: Optional[type] = None, *, meta=DataClassMeta, **options) -> T
     :key order: Generate comparison methods other than __eq__
     :key unsafe_hash: Force generation of __hash__
     :key frozen: Allow field reassignment after initialisation
-    :key hide_internals: Hide internal methods in __repr__
+    :key hide_internals: Hide internal methods in __repr__, comparison, and __iter__
```
Comment (biqqles) [inline dataclassy/decorator.py:24]:
```suggestion
    :key hide_internals: Hide internal methods in __repr__, __iter__ and comparison methods
```

**Review by biqqles — APPROVED** [review]
I'm really sorry about the delay but grateful for this patch

**Comment by thisisrandy** [comment]
Not a problem. Thanks for getting to it!

**Pull request merged**